### DISCUSSION TITLE
JDBI 3.37.0

### DISCUSSION THREAD

**Discussion by hgschmie** [jdbi/jdbi#2272] [Announcements]
** DO NOT USE. PLEASE UPGRADE TO 3.37.1 **

- upgrade to geantyref 1.3.14
- removes the core dependency on the caffeine library. This now uses a simple LRU cache for sql parser and sql statements.
- adds pluggable cache implementation using caffeine. The old caching behavior can now be restored by using the `jdbi3-caffeine-cache` dependency and adding `jdbi.installPlugin(new CaffeineCachePlugin());`.
- adds pluggable no-op cache implementation for testing and debugging
- improve PostGIS null value handling (#2260, thanks @tprebs)
- use Postgres 15 (native arm64) for building on MacOS arm64

<hr /><em>This discussion was created from the release <a href='https://github.com/jdbi/jdbi/releases/tag/v3.37.0'>JDBI 3.37.0</a>.</em>

**Comment by jagerem** [comment]
Very Welcome,
 I did not find more convinient explanation an about the dependency injection as is in the documentation which look like this: 
class InjectedDependencies implements JdbiConfig<InjectedDependencies> {
    @Inject - what package this annotation comes from ?
    SomeDependency dep;

    public InjectedDependencies() {} - is Ok.

    @Override
    public InjectedDependencies createCopy() { - is Ok.
        return this; // effectively immutable
    }
}

Jdbi jdbi = Jdbi.create(myDataSource); - is Ok.
myIoC.inject(jdbi.getConfig(InjectedDependencies.class)); - what is the myIoc.inject ?

// Then, in any component that needs to access it:
getHandle().getConfig(InjectedDependencies.class).dep - this is ok!

Could you please clarify how to implement independency injection properly ? BTW - jdbi is very OK and is  what I have been looking for - becouse I am tired of Hibernate and Spring and MyBatis is ok but not for me.
Best regards
Jarek
P.S. Thank you in advance for any guidance.

**Reply by hgschmie** [reply]
Hi Jarek, 

Which injection code do you refer to? I assume you look at the chapter 19.1 (Simple dependency injection).
`@Inject` is  standard annotation, either use the JSR-330 annotation (`javax.inject.Inject`) or the CDI ones (`jakarta.inject.Inject`), depending on the IoC container that you use. The agnostic code supports most containers (Spring Framework, Google Guice or the JavaEE 6+ CDI container). 

If you are looking for more comprehensive examples, there is the Google Guice support (chapter 18.7) or Spring 5+ support (18.13).
